Style Guide and Diataxis Review - Chrysalis III (#117)
iotaledger/chrysalis-docs: docs/guides/migration_mechanism.md

- explanation
---

# How the IOTA Chrysalis Phase 2 Token Migration Works

For the Chrysalis Phase 2 transition, the IOTA Foundation developed a way to migrate funds to the new network. IOTA created this new mechanism because the previous network did not support the WOTS addresses.

Before proceeding into the developed mechanism, below is our reasoning for why other mechanisms were not chosen:

- `Migrating funds by performing periodic global snapshots on both networks`: while this approach would have
  been the easiest to do, it would have blocked the funds of people between the times of global snapshots. For example, if
  we would have taken global snapshots monthly (to carry over funds from burn/migration addresses), token holders would
  have been excluded from opportunities arising from the change in token price. In this case, perhaps a user wanted to sell
  tokens but were blocked by the fact that the next global snapshot would only be performed in 20 days and so they
  couldn't move their tokens to an exchange (which only supports Chrysalis Phase 2 IOTA).
- `Supporting WOTS on Chrysalis Phase 2 (and only allowing to send to non-WOTS addresses)`: while this was also a

>   viable option, we decided not to include support for WOTS as it brought several legacy problems we did not want, including:

Reviewer: ```suggestion
  viable option, we decided not to include support for WOTS as it brought several legacy problems:
```

Author: 👍

- Firefly
- wallet
- legacy nodes
- explanation
---

# How the IOTA Chrysalis Phase 2 Token Migration Works

> For the Chrysalis Phase 2 transition, the IOTA Foundation had to develop a way for funds to be migrated to the new network. Because the previous network did not support the WOTS addresses, this new mechanism was created.

Reviewer 2: ```suggestion
For the Chrysalis Phase 2 transition, the IOTA Foundation developed a way to migrate funds to the new network. IOTA created this new mechanism because the previous network did not support the WOTS addresses.
```

> The developed mechanism includes that:

Reviewer 2: I'm not sure about this phrase and the following bullet points.

Author: Agreed

- `Migrating funds by performing periodic global snapshots on both networks`: while this approach would have
  been the easiest to do, it would have blocked the funds of people between the times of global snapshots. For example, if
  we would have taken global snapshots monthly (to carry over funds from burn/migration addresses), token holders would
  have been excluded from opportunities arising from the change in token price. In this case, perhaps a user wanted to sell
  tokens but were blocked by the fact that the next global snapshot would only be performed in 20 days and so they
  couldn't move their tokens to an exchange (which only supports Chrysalis Phase 2 IOTA).
- `Supporting WOTS on Chrysalis Phase 2 (and only allowing to send to non-WOTS addresses)`: while this was also a
  viable option, we decided not to include support for WOTS as it brought several legacy problems we did not want, including:
    - WOTS signatures are very large and make up a disproportional amount of data in a transaction (note that our PoW
      requirement in Chrysalis' Phase 2 was dependent on the size of the message). Additionally, there were no real bounds

>       on how big such signatures could grow to (even if, per the default, we only supported 3 security levels in our

Reviewer 2: ```suggestion
      on how big such signatures could grow to (even if, per the default, we only supported three security levels in our
```

      accessible from.
    - A checksum of that Ed25519 address.
    - A tryte prefix `TRANSFER` (these addresses always start with this prefix).

> 2. As mentioned above, a `Receipt` could only be contained in a milestone and therefore the Coordinator on the Chrysalis

Reviewer 2: ```suggestion
2. As mentioned above, a `Receipt` could only be contained in a milestone and therefore the Coordinator on the Chrysalis.
```
